natlib/svcd.c

```diff
@@ -185,3 +192,201 @@ static int svcd_init( lua_State *L )
+    
+    /* SET IVKID */
+    resolve_table(L, "ikvid"); //7
+    int ivkid = (int) lua_tonumber(L, -1);
+   
+    printf("Getting ikvid: %i\n", ivkid);
+  
+    ivkid =  ivkid + 1;
+    if( ivkid > 65535 ) { 
+        ivkid = 0;
+    }
+
+    lua_settop(L, numargs);
+
+    printf("Updating ikvid to %i\n", ivkid);
+    // setting ivkid
+    lua_getglobal(L, "SVCD");
+    lua_pushnumber(L, ivkid);
+    lua_setfield(L, -2, "ikvid");
+    lua_settop(L, numargs);
+    /* END SET IVKID*/
+ 
+    /* SET HANDLER */
+    printf("Setting handler\n");
+    resolve_table(L, "handlers"); // 7
+    printf("Resolved handlers\n");
+
+    lua_pushnumber(L, ivkid);
+    printf("Resolved ikvid handler\n");
+    lua_pushlightfunction(L, on_done);
+
+    lua_settable(L, -3);
+    /* END SET HANDLER */
+
+    /* INVOKE LATER */
+    printf("Setting up invoke later \n");
+    
+    /* CUSTOM WRITE INVOKE */
+    // lua_getglobal(L, "scvd_write_invoke_later");
+    // lua_pushnumber(L, ivkid);
+    // lua_pushnumber(L, timeout_ms);
+    // printf("Calling invoke later custom \n");
+    // lua_call(L, 2, 0);
+    /* END CUSTOM */
+
+    
+    /* LUA WRITE INVOKE */
+    resolve_table(L, "write_invoke_later"); // 9
+     //function write invoke later 
+        lua_pushnumber(L, ivkid);  // 10
+        lua_pushnumber(L, timeout_ms);
+        printf("Calling invoke later \n");
+        lua_call(L, 2, 0);
+    /* END LUA WRITE INVOKE */
+
+
+    /* SEND TO */
+    printf("Setting up sendto \n");
+    lua_pushlightfunction(L, libstorm_net_sendto); 
+    // resolve_table(L, "print_sendto"); 
+        resolve_table(L, "wcsock");
+        /* BEGIN MESSAGE PACK */
+        printf("Setting up msg_pack array \n");
+        lua_pushlightfunction(L, libmsgpack_mp_pack);
+            lua_newtable(L);
+            //svcid
+            lua_pushnumber(L, 1);
+            lua_pushnumber(L, svcid);
+            lua_settable(L, -3);
+            printf("Appending svcid 0x%x\n", svcid);
+            //attrid
+            lua_pushnumber(L, 2);
+            lua_pushnumber(L, attrid);
+            lua_settable(L, -3);
+            printf("Appending attrid 0x%x\n", attrid);
+
+            // ivkid
+            lua_pushnumber(L, 3);
+            lua_pushnumber(L, ivkid); //14
+            lua_settable(L, -3);
+            printf("Appending ivkid 0x%x\n", ivkid);
+
+            //payload
+            lua_pushnumber(L, 4);
+            lua_pushlstring(L, payload, l);
+            lua_settable(L, -3);
+            printf("Appending payload %s\n", payload);
+
+        printf("Calling msg_pack \n");
+        lua_call(L, 1, 1);
+
+        /* END OF MSG PACK FUNCTION */
+
+        // targetip    
+        lua_pushlstring(L, targetip, g);
+        //2526
+        lua_pushnumber(L, 2526);
+
+    printf("Calling sendto\n");
+    lua_call(L, 4, 0);
+    /* END SEND_TO */
+
+    return 0; //return # of arguments
+}
+
+
```

Comment:
Good job on writing the function. I think it would be better to remove commented code, print statements, etc from the code before we can merge it into master. 

Untitled